app/javascript/components/lesson-preview/index.jsx

```diff
@@ -8,9 +9,7 @@ import LessonContentInput from './components/lesson-content-input';
 import LessonContentPreview from './components/lesson-content-preview';
 import axios from '../../src/js/axiosWithCsrf';
 
-import { generateLink, encodeContent, decodeContent } from '../../src/js/previewShare';
-
-const LessonPreview = () => {
+const LessonPreview = ({previewContent}) => {
   const [content, setContent] = useState('');
```

Comment:
We can remove the useEffect for setting the content if the query is present and just set the previewContent as the initial content. It should be an empty string if it's empty.
```suggestion
  const [content, setContent] = useState(previewContent);
```